internal/services/policy/remediation_resource.go

```diff
@@ -84,49 +83,53 @@ func resourceArmResourcePolicyRemediation() *pluginsdk.Resource {
 			"resource_discovery_mode": {
 				Type:     pluginsdk.TypeString,
 				Optional: true,
-				Default:  string(policyinsights.ExistingNonCompliant),
+				Default:  string(policyinsights.ResourceDiscoveryModeExistingNonCompliant),
 				ValidateFunc: validation.StringInSlice([]string{
-					string(policyinsights.ExistingNonCompliant),
-					string(policyinsights.ReEvaluateCompliance),
+					string(policyinsights.ResourceDiscoveryModeExistingNonCompliant),
+					string(policyinsights.ResourceDiscoveryModeReEvaluateCompliance),
 				}, false),
 			},
 		},
 	}
 }
 
 func resourceArmResourcePolicyRemediationCreateUpdate(d *pluginsdk.ResourceData, meta interface{}) error {
-	client := meta.(*clients.Client).Policy.RemediationsClient
+	client := meta.(*clients.Client).Policy.PolicyInsightsClient
 	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
 	defer cancel()
 
 	resourceId := d.Get("resource_id").(string)
 
-	id := parse.NewResourcePolicyRemediationId(resourceId, d.Get("name").(string))
+	id := policyinsights.NewScopedRemediationID(resourceId, d.Get("name").(string))
 
 	if d.IsNewResource() {
-		existing, err := client.GetAtResource(ctx, id.ResourceId, id.Name)
+		existing, err := client.RemediationsGetAtResource(ctx, id)
 		if err != nil {
-			if !utils.ResponseWasNotFound(existing.Response) {
+			if !response.WasNotFound(existing.HttpResponse) {
 				return fmt.Errorf("checking for presence of existing %s: %+v", id.ID(), err)
 			}
 		}
-		if existing.ID != nil && *existing.ID != "" {
-			return tf.ImportAsExistsError("azurerm_resource_policy_remediation", *existing.ID)
+		if existing.Model != nil && existing.Model.Id != nil && *existing.Model.Id != "" {
+			return tf.ImportAsExistsError("azurerm_resource_policy_remediation", *existing.Model.Id)
 		}
 	}
 
 	parameters := policyinsights.Remediation{
-		RemediationProperties: &policyinsights.RemediationProperties{
+		Name: utils.String(id.RemediationName),
+		Properties: &policyinsights.RemediationProperties{
 			Filters: &policyinsights.RemediationFilters{
 				Locations: utils.ExpandStringSlice(d.Get("location_filters").([]interface{})),
 			},
-			PolicyAssignmentID:          utils.String(d.Get("policy_assignment_id").(string)),
-			PolicyDefinitionReferenceID: utils.String(d.Get("policy_definition_id").(string)),
-			ResourceDiscoveryMode:       policyinsights.ResourceDiscoveryMode(d.Get("resource_discovery_mode").(string)),
+			PolicyAssignmentId:          utils.String(d.Get("policy_assignment_id").(string)),
+			PolicyDefinitionReferenceId: utils.String(d.Get("policy_definition_id").(string)),
 		},
 	}
+	if v := d.Get("resource_discovery_mode").(string); v != "" {
```

Comment:
Since `resource_discovery_mode` has a default set in the schema, it means it will always be a non empty string, this `if` is not necessary here.